pkg/verifier/aws/entry_point.go

```diff
@@ -101,55 +100,33 @@ func (a *AwsVerifier) ValidateEgress(vei verifier.ValidateEgressInput) *output.O
 			a.Output.AddError(err)
 		}
 
-		//Check if a keypair was uploaded
-		searchKeys := []string{DEBUG_KEY_NAME}
+		// Check if a keypair was uploaded
+		searchKeys := []string{DebugKeyName}
 		_, err := a.AwsClient.DescribeKeyPairs(vei.Ctx, &ec2.DescribeKeyPairsInput{
 			KeyNames: searchKeys,
 		})
 		if err != nil {
-			//if no key was found continue without executing deletion code
-			fmt.Printf("Debug KeyPair %v not found \n", DEBUG_KEY_NAME)
+			// If no key was found continue without executing deletion code
+			fmt.Printf("Debug KeyPair %v not found \n", DebugKeyName)
 		} else {
-			//if there was a key found, then delete it.
+			// If there was a key found, then delete it.
 			_, err = a.AwsClient.DeleteKeyPair(vei.Ctx, &ec2.DeleteKeyPairInput{
-				KeyName: awsTools.String(DEBUG_KEY_NAME),
+				KeyName: awsTools.String(DebugKeyName),
 			})
-			//if there was any issues deleting the keypair.
+			// If there were any issues deleting the keypair.
 			if err != nil {
 				a.Output.AddError(err)
 			}
-
 		}
 
 		return &a.Output
 	}
 
-	// Fetch the egress URL list from github, falling back to local lists in the event of a failure.
-	// Note that this is TOTALLY IGNORED by LegacyProbe,
-	// as that probe only knows how to use the egress URL lists baked into its
-	// AMIs/container images
-	egressListYaml := vei.EgressListYaml
-	var egressListStr, tlsDisabledEgressListStr string
-	if egressListYaml == "" {
-		githubEgressList, err := egress_lists.GetGithubEgressList(vei.PlatformType)
-		if err != nil {
-			a.Logger.Error(vei.Ctx, "Failed to get egress list from GitHub, falling back to local list: %v", err)
-
-			egressListYaml, err = egress_lists.GetLocalEgressList(vei.PlatformType)
-			if err != nil {
-				return a.Output.AddError(err)
-			}
-		} else {
-			egressListYaml, err = githubEgressList.GetContent()
-			if err != nil {
-				return a.Output.AddError(err)
-			}
-
-			a.Logger.Info(vei.Ctx, "Using egress URL list from %s at SHA %s", githubEgressList.GetURL(), githubEgressList.GetSHA())
-		}
-	}
+	// Generate both egress lists for the given PlatformType. Note: the result of this is ignored by the Legacy probe.
+	generatorVariables := map[string]string{"AWS_REGION": a.AwsClient.Region}
+	generator := egress_lists.NewGenerator(vei.PlatformType, generatorVariables, a.Logger)
 
-	egressListStr, tlsDisabledEgressListStr, err = egress_lists.EgressListToString(egressListYaml, map[string]string{"AWS_REGION": a.AwsClient.Region})
+	egressListStr, tlsDisabledEgressListStr, err := generator.GenerateEgressLists(vei.Ctx, vei.EgressListYaml)
```

Comment:
The extraction of the fetching login into `egress_lists` essentially allows us to have this common call flow across AWS + GCP, and reads more cleanly in the scope of this already large function.